html2asketch/textStyle.js

```diff
@@ -32,17 +32,25 @@ class TextStyle {
   }
 
   toJSON() {
-    return {
+    const json = {
       'color': this._color,
       'fontSize': this._fontSize,
       'fontFamily': this._fontFamily,
       'fontWeight': this._fontWeight,
-      'lineHeight': this._lineHeight,
-      'letterSpacing': this._letterSpacing,
       'textTransform': this._textTransform,
       'textDecoration': this._textDecoration,
       'textAlign': this._textAlign
     };
+
+    if (this._lineHeight !== undefined) {
+      json['lineHeight'] = this._lineHeight;
+    }
+
+    if (this._letterSpacing !== undefined) {
+      json['letterSpacing'] = this._letterSpacing;
+    }
```

Comment:
Have you figured out how this is different from passing `undefined` to the property? As far as I can see `JSON.stringify({test:undefined})` produces `{}` so this code shouldn't be changing anything 🤔

Reply:
Nope, haven't figured that out yet. Is that a blocker for this PR?

Reply:
I test this same code on both puppeteer 0.9.0 and 0.12.0.

```javascript
const puppeteer = require('puppeteer');


puppeteer.launch().then(async browser => {

  const page = await browser.newPage();
  await page.setContent('<script>const json = {test: undefined}</script>');

  const json = await page.evaluate('json');
  console.log(json);

  await browser.close();

});
```

0.9.0 output is `{'test': null}`,
0.12.0 output is `{}`.

I think a `page.evaluate` in puppeteer is causing problems.

Reply:
Thanks @KimDal-hyeong for investigating this! If that's a puppeteer issue, there is no reason for us to merge this change IMHO. @markdalgleish can you please verify if that works for your setup?

Reply:
Yeah, it looks like the issue goes away if I call `JSON.stringify` in a Puppeteer context, rather than passing an object reference across the bridge. I'm using Puppeteer v0.13.0 by the way, which is the latest version right now.

Reply:
Seems we've traded one set of nulls for another.

With the latest version, even when stringifying inside Puppeteer, the `fontSize` and `letterSpacing` nulls have gone, but I'm now getting nulls for some `x` and `y` values.

e.g. `{"_class":"rect","constrainProportions":false,"height":0,"width":0,"x":null,"y":null}`